### PR TITLE
RCM 103 - Backport of adding death date to Cohort builder to 1.x 

### PR DESCRIPTION
**Pull request by ssmusoke**
…s while maintaining the old API methods

(cherry picked from commit 2ea48e54a90dbbf525c6b7972f9bc0f99b87e38f)

### CODE REVIEW

**Comment by ssmusoke** [comment]
@dkayiwa Having issues running the tests so please help - I have backported the changes from master, and resolved compile issues

**Comment by dkayiwa** [comment]
@ssmusoke i thought you are already running platform 2.x
If so, why do you need versions 1.x of this module?

**Comment by ssmusoke** [comment]
@dkayiwa the 2.x based version is not yet released there are sites running
the last 1.x release

**Comment by ssmusoke** [comment]
@dkayiwa ping

File: omod/src/main/resources/moduleApplicationContext.xml [hunk 1]
```diff
@@ -27,6 +27,48 @@
 		</property>
 	</bean>
 	
+	<bean id="reportService" parent="serviceContext">
```
Comment (ssmusoke) [inline omod/src/main/resources/moduleApplicationContext.xml]: @dkayiwa This looks like the major issue - not sure why it was configured like this in the 2.x branch

**Comment by ssmusoke** [comment]
@dkayiwa The service classes are needed... 

1. PatientCharacteristicFilter invokes the ReportService, which the implementation is required

2. The ReportDAO is where the filters are applied

3. The ReportConstants is used in the service classes. I am brining the cherry picks from the 2.x classes so that I do not have to figure out the flows again - they should just work, or not?

**Comment by dkayiwa** [comment]
Cherry picks will not work here because the two branches are very different.
You will need to manually put the changes in the 1.x branch one by one. But only the required ones. For instance, i have not seen a reason why you need to touch the module spring application context file if all you are adding is support for death date. Can't you just add these two to the existing services classes?

**Comment by ssmusoke** [comment]
@dkayiwa Got you, closing this PR and opening a new one

**Pull request closed without merging**